software/controller/lib/hal/system_timer.cpp

```diff
@@ -42,7 +42,7 @@ void SystemTimer::initialize(uint32_t cpu_frequency_MHz) {
   // -1 until the clock wraps back to zero and generates an interrupt. This
   // setting will cause an interrupt every 10,000 clocks or 1 millisecond
   tmr->auto_reload = 9999;
-  tmr->prescaler = (cpu_frequency_MHz / 10 - 1);
+  tmr->prescaler = static_cast<uint32_t>(cpu_frequency.megahertz() / 10.0f - 1.0f);
```

Comment:
Should this be an explicit rounding, floor or ceiling? Not clear which is intended. May be a few other instances of something like this that I missed.